src/main/java/net/minecraftforge/client/extensions/ForgeItemBlockRenderTypes.java

```diff
@@ -0,0 +1,103 @@
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+public class ForgeItemBlockRenderTypes
+{
+    private static final Predicate<RenderType> SOLID_PREDICATE = type -> type == RenderType.solid();
+    private static final Object LOCK = new Object();
+
+    private static final Map<Holder.Reference<Block>, Predicate<RenderType>> blockRenderChecks = createMap();
+    private static final Map<Holder.Reference<Fluid>, Predicate<RenderType>> fluidRenderChecks = createMap();
+
+    @Nullable
+    private static volatile Map<Holder.Reference<Block>, Predicate<RenderType>> blockRenderChecksReadOnly = null;
+    @Nullable
+    private static volatile Map<Holder.Reference<Fluid>, Predicate<RenderType>> fluidRenderChecksReadOnly = null;
+
+    public static void setRenderLayer(Block block, Predicate<RenderType> predicate)
+    {
+        Objects.requireNonNull(predicate);
+        synchronized (LOCK)
+        {
+            blockRenderChecks.put(ForgeRegistries.BLOCKS.getDelegateOrThrow(block), predicate);
+            blockRenderChecksReadOnly = null;
+        }
+    }
+
+    public static void setRenderLayer(Fluid fluid, Predicate<RenderType> predicate)
+    {
+        Objects.requireNonNull(predicate);
+        synchronized (LOCK)
+        {
+            fluidRenderChecks.put(ForgeRegistries.FLUIDS.getDelegateOrThrow(fluid), predicate);
+            fluidRenderChecksReadOnly = null;
+        }
+    }
+
+    protected static Map<Holder.Reference<Block>, Predicate<RenderType>> getBlockLayerPredicates()
+    {
+        Map<Holder.Reference<Block>, Predicate<RenderType>> map = blockRenderChecksReadOnly;
+        if (map == null)
+        {
+            synchronized (LOCK)
+            {
+                blockRenderChecksReadOnly = map = copy(blockRenderChecks);
+            }
+        }
+        return map;
+    }
+
+    protected static Map<Holder.Reference<Fluid>, Predicate<RenderType>> getFluidLayerPredicates()
+    {
+        Map<Holder.Reference<Fluid>, Predicate<RenderType>> map = fluidRenderChecksReadOnly;
+        if (map == null)
+        {
+            synchronized (LOCK)
+            {
+                fluidRenderChecksReadOnly = map = copy(fluidRenderChecks);
+            }
+        }
+        return map;
+    }
+
+    private static <T> Map<Holder.Reference<T>, Predicate<RenderType>> copy(Map<Holder.Reference<T>, Predicate<RenderType>> map)
+    {
+        var newMap = new Object2ObjectOpenHashMap<>(map);
+        newMap.defaultReturnValue(SOLID_PREDICATE);
+        return newMap;
+    }
+
+    private static <T> Map<Holder.Reference<T>, Predicate<RenderType>> createMap()
+    {
+        return Util.make(new Object2ObjectOpenHashMap<>(Object2ObjectOpenHashMap.DEFAULT_INITIAL_SIZE, .75F), map ->
+                map.defaultReturnValue(SOLID_PREDICATE)
+        );
```

Comment:
This is more a matter of style, but should we use `Util.make` here? To my knowledge, that method is designed for use in field initializers which need to modify the value further and don't wish to use a `static` class initializer block due to ugliness, such as adding entries to a modifiable map (or configuring a map, in this case). 

In my opinion, we can (and should) do this without `Util.make`, as we already do in `#copy` above.

Reply:
I'll get rid of it, it's a leftover from copying this back and forth between a method and a field initializer.